src/ol/layer/layer.js

```diff
@@ -172,6 +172,7 @@ ol.layer.Layer.prototype.setMap = function(map) {
         map, ol.render.EventType.PRECOMPOSE, function(evt) {
           var layerState = this.getLayerState();
           layerState.managed = false;
+          layerState.zIndex = 1000;
```

Comment:
I've thought about that too. But using a non-infinity value makes things more flexible to the user, as it is still possible to draw layers above unmanaged layers.

Also, with @gberaudo's current patch, calling `setZIndex` on an unmanaged layer has no effect. I think this should at least me documented. But it may also be great to support calling `setZIndex` on an unmanaged layer. An unmanaged layer would then just be a layer that the map doesn't know about (so you can silently add and remove layers if you want). Maybe I'm going to far here. And if people think I am indeed going too far then I'll go with `Infinity` as suggested by Tim.